model/src/http/interaction.rs

```diff
@@ -0,0 +1,169 @@
+//! Models used when responding to interactions over HTTP.
+
+use crate::{
+    application::{command::CommandOptionChoice, component::Component},
+    channel::{
+        embed::Embed,
+        message::{AllowedMentions, MessageFlags},
+    },
+    http::attachment::Attachment,
+};
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+/// Interaction response sent to Discord.
+///
+/// See [Discord Docs/Interaction Object].
+///
+/// [Discord Docs/Interaction Object]: https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-interaction-structure
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct InteractionResponse {
+    /// Type of the response.
+    #[serde(rename = "type")]
+    pub kind: InteractionResponseType,
+    /// Data of the response.
+    ///
+    /// This is required if the type is any of the following:
+    /// - [`ChannelMessageWithSource`]
+    /// - [`UpdateMessage`]
+    /// - [`ApplicationCommandAutocompleteResult`]
+    ///
+    /// [`ApplicationCommandAutocompleteResult`]: InteractionResponseType::ApplicationCommandAutocompleteResult
+    /// [`ChannelMessageWithSource`]: InteractionResponseType::ChannelMessageWithSource
+    /// [`UpdateMessage`]: InteractionResponseType::UpdateMessage
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub data: Option<InteractionResponseData>,
+}
+
+/// Data included in an interaction response.
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct InteractionResponseData {
+    /// Allowed mentions of the response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub allowed_mentions: Option<AllowedMentions>,
+    /// List of attachments on the response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub attachments: Option<Vec<Attachment>>,
+    /// List of autocomplete alternatives.
+    ///
+    /// Can only be used with
+    /// [`InteractionResponseType::ApplicationCommandAutocompleteResult`].
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub choices: Option<Vec<CommandOptionChoice>>,
+    /// List of components on the response.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub components: Option<Vec<Component>>,
+    /// Content of the response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub content: Option<String>,
+    /// Embeds of the response.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub embeds: Option<Vec<Embed>>,
+    /// Interaction response data flags.
+    ///
+    /// The only supported flag is [`MessageFlags::EPHEMERAL`].
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub flags: Option<MessageFlags>,
```

Comment:
`SUPPRESS_EMBEDS` can also be set cf. https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-response-object-messages

Reply:
I'll make this change here and also for now a docs note targeting `main`.